dashpay/src/main/kotlin/org/dashevo/dashpay/ContactRequests.kt

```diff
@@ -21,16 +21,18 @@ class ContactRequests(val platform: Platform) {
         val contactKey = contactKeyChain.watchingKey
         val contactPub = contactKey.serializeContactPub()
 
-        val encryptedContactPubKey = fromUser.encryptExtendedPublicKey(contactPub, toUser, 0, aesKey)
-        val xpubBase64 = Base64.toBase64String(encryptedContactPubKey)
+        val (encryptedContactPubKey, encryptedAccountLabel) = fromUser.encryptExtendedPublicKey(contactPub, toUser, 0, aesKey)
+        val accountReference = fromUser.getAccountReference(aesKey, toUser)
         val timeStamp = Date().time
 
         val contactRequestDocument = platform.documents.create(
             CONTACTREQUEST_DOCUMENT, fromUser.uniqueIdString, mutableMapOf<String, Any?>(
-                "encryptedPublicKey" to xpubBase64,
+                "encryptedPublicKey" to encryptedContactPubKey,
                 "recipientKeyIndex" to toUser.publicKeys[0].id,
                 "senderKeyIndex" to fromUser.identity!!.publicKeys[0].id,
-                "toUserId" to toUser.id,
+                "toUserId" to Base58.decode(toUser.id),
+                "accountReference" to accountReference,
+                //"encryptedAccountLabel" to encryptedAccountLabel,  //contact requires 64 bytes, but minimum should be 32
```

Comment:
This field has a bug in the dashpay contract, so we will not include it for now.

Reply:
Can we please add a TODO so we can remember when see this code again?

Reply:
https://github.com/dashevo/android-dashpay/pull/23/commits/0962a8b8b39c186520d0c30f060553d12b8fbee3 fixed here.